File: client/java/src/avt/HomeMsgHandler.java
package avt;

import java.util.Vector;
import main.Canvas;

public final class HomeMsgHandler extends IService implements IMiniGameMsgHandler {
   public static HomeMsgHandler instance = new HomeMsgHandler();

   public static void onHandler() {
      GlobalMessageHandler.gI().miniGameMessageHandler = instance;
   }

   public final void onMessage(Message var1) {
      try {
         int var2;
         int var7;
         short var17;
         Vector var18;
         short var19;
         switch (var1.command) {
            case -75:
               int var12 = var1.reader().readInt();
               Canvas.inputDlg.setImg(T.setPass, new IActionSetPass(this, var12), 0);
            default:
               return;
            case -73:
               var19 = var1.reader().readShort();
               byte[] var22 = new byte[var1.reader().readInt()];
               var1.reader().read(var22);
               HouseScr.gI().saveTileMap((byte[])var22, var19);
               return;
            case -67:
               byte var14 = var1.reader().readByte();
               byte var16 = -1;
               var17 = 0;
               var18 = null;
               if (var14 == 0) {
                  var17 = var1.reader().readShort();
                  var16 = var1.reader().readByte();
               } else {
                  var18 = new Vector();
                  var19 = var1.reader().readShort();

                  for(var7 = 0; var7 < var19; ++var7) {
                     Avatar var21;
                     (var21 = new Avatar()).IDDB = var1.reader().readInt();
                     var21.typeHome = var1.reader().readByte();
                     var18.addElement(var21);
                  }
               }

               HouseScr.gI().onGetTypeHouse(var14, var16, var17, var18);
               return;
            case -66:
               MapItem var13;
               (var13 = new MapItem()).typeID = var1.reader().readShort();
               var13.x = var1.reader().readByte();
               var13.y = var1.reader().readByte();
               HouseScr.gI().onRemoveItem(var13);
               return;
            case -65:
               var17 = (short)var1.reader().readByte();
               var2 = var1.reader().readInt();
               short var5;
               short[] var15 = new short[var5 = var1.reader().readShort()];

               int var8;
               for(var8 = 0; var8 < var5; ++var8) {
                  var15[var8] = (short)var1.reader().readByte();
               }

               var19 = (short)var1.reader().readByte();
               var18 = new Vector();
               var7 = var1.reader().readShort();

               for(var8 = 0; var8 < var7; ++var8) {
                  MapItem var23;
                  (var23 = new MapItem()).typeID = var1.reader().readShort();
                  var23.x = var1.reader().readByte() * 24;
                  var23.y = var1.reader().readByte() * 24;
                  var23.dir = var1.reader().readByte();
                  var18.addElement(var23);
               }

               Vector var20 = GlobalMessageHandler.readListPlayer(var1);
               ParkMsgHandler.onHandler();
               HouseScr.gI().onJoin((byte)var17, var2, var15, (byte)var19, var18, var20);
               return;
            case -46:
               short var9 = var1.reader().readShort();
               String var11 = var1.reader().readUTF();
               HouseScr.gI().onCreateHome(var9, var11);
               return;
            case -43:
               Tile[] var3 = new Tile[var2 = var1.reader().readShort()];

               for(int var4 = 0; var4 < var2; ++var4) {
                  var3[var4] = new Tile();
                  var3[var4].name = var1.reader().readUTF();
                  var3[var4].priceXu = var1.reader().readInt();
                  var3[var4].priceLuong = var1.reader().readInt();
               }

               HouseScr.gI().onGetTileInfo(var3);
               return;
            case 51:
               MapScr.gI().onPlayerJoinPark(ParkMsgHandler.playerJoinBoard(var1));
               return;
            case 76:
               GlobalMessageHandler.readMove(var1);
               return;
            case 77:
               GlobalMessageHandler.readChat(var1);
         }
      } catch (Exception var19) {
         var19.printStackTrace();
      }

   }
}

File: client/unity/Assets/Scripts/Assembly-CSharp/HomeMsgHandler.cs
using System;

public class HomeMsgHandler : IMiniGameMsgHandler
{
	private class IActionSetPass : IKbAction
	{
		private int idUser5;

		public IActionSetPass(int idUser)
		{
			idUser5 = idUser;
		}

		public void perform(string text)
		{
			AvatarService.gI().doSetPassMyHouse(text, idUser5, 1);
			Canvas.endDlg();
		}
	}

	private static HomeMsgHandler instance = new HomeMsgHandler();

	public static void onHandler()
	{
		GlobalMessageHandler.gI().miniGameMessageHandler = instance;
	}

	public void onMessage(Message msg)
	{
		try
		{
			switch (msg.command)
			{
			case -43:
			{
				short num6 = msg.reader().readShort();
				Tile[] array2 = new Tile[num6];
				for (int l = 0; l < num6; l++)
				{
					array2[l] = new Tile();
					array2[l].name = msg.reader().readUTF();
					array2[l].priceXu = msg.reader().readInt();
					array2[l].priceLuong = msg.reader().readInt();
				}
				HouseScr.gI().onGetTileInfo(array2);
				break;
			}
			case -65:
			{
				sbyte houseType2 = msg.reader().readByte();
				int idUser2 = msg.reader().readInt();
				short[] array = null;
				short num4 = msg.reader().readShort();
				array = new short[num4];
				for (int j = 0; j < num4; j++)
				{
					array[j] = msg.reader().readByte();
				}
				sbyte w = msg.reader().readByte();
				MyVector myVector2 = new MyVector();
				short num5 = msg.reader().readShort();
				for (int k = 0; k < num5; k++)
				{
					MapItem mapItem = new MapItem();
					mapItem.typeID = msg.reader().readShort();
					mapItem.x = (mapItem.xTo = msg.reader().readByte() * 24);
					mapItem.y = (mapItem.yTo = msg.reader().readByte() * 24);
					mapItem.dir = msg.reader().readByte();
					myVector2.addElement(mapItem);
				}
				MyVector listPlayer = GlobalMessageHandler.readListPlayer(msg);
				ParkMsgHandler.onHandler();
				HouseScr.gI().onJoin(houseType2, idUser2, array, w, myVector2, listPlayer);
				break;
			}
			case 51:
				ParkMsgHandler.playerJoinBoard(msg);
				break;
			case 54:
				GlobalMessageHandler.readMove(msg);
				break;
			case 55:
				GlobalMessageHandler.readChat(msg);
				break;
			case -46:
			{
				short type = msg.reader().readShort();
				string str = msg.reader().readUTF();
				HouseScr.gI().onCreateHome(type, str);
				break;
			}
			case -66:
			{
				MapItem mapItem2 = new MapItem();
				mapItem2.typeID = msg.reader().readShort();
				mapItem2.x = msg.reader().readByte();
				mapItem2.y = msg.reader().readByte();
				HouseScr.gI().onRemoveItem(mapItem2);
				break;
			}
			case -67:
			{
				int num2 = msg.reader().readByte();
				int houseType = -1;
				short verTile = 0;
				MyVector myVector = null;
				if (num2 == 0)
				{
					verTile = msg.reader().readShort();
					houseType = msg.reader().readByte();
				}
				else
				{
					myVector = new MyVector();
					short num3 = msg.reader().readShort();
					for (int i = 0; i < num3; i++)
					{
						Avatar avatar = new Avatar();
						avatar.IDDB = msg.reader().readInt();
						avatar.typeHome = msg.reader().readByte();
						myVector.addElement(avatar);
					}
				}
				HouseScr.gI().onGetTypeHouse(num2, houseType, verTile, myVector);
				break;
			}
			case -73:
			{
				int wNum = msg.reader().readShort();
				int num = msg.reader().readInt();
				sbyte[] data = new sbyte[num];
				msg.reader().read(ref data);
				HouseScr.gI().saveTileMap(data, wNum);
				break;
			}
			case 76:
				GlobalMessageHandler.readMove(msg);
				break;
			case 77:
				GlobalMessageHandler.readChat(msg);
				break;
			case -75:
			{
				int idUser = msg.reader().readInt();
				ipKeyboard.openKeyBoard(T.setPass, ipKeyboard.PASS, string.Empty, new IActionSetPass(idUser), false);
				break;
			}
			}
		}
		catch (Exception e)
		{
			Out.logError(e);
		}
	}
}
